Source/Applications/Web/AllApplicationsQuery.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+import { BehaviorSubject } from 'rxjs';
+import { ObservableQuery, NetworkStatus } from 'apollo-client';
+import gql from 'graphql-tag';
+import { injectable, singleton } from 'tsyringe';
+import { ApplicationModel } from './ApplicationModel';
+import { DataSource } from './DataSource';
+
+@singleton()
+@injectable()
+export class AllApplicationsQuery {
+    items: BehaviorSubject<ApplicationModel[]> = new BehaviorSubject<ApplicationModel[]>(
+        []
+    );
+    private _observableQuery?: ObservableQuery<any>;
+
+    constructor(readonly _dataSource: DataSource) {
+        this.startPollingForApplications();
+    }
+
+    public startPollingForApplications() {
+        this._startPollingForApplications();
+    }
+
+    public stopPollingForApplications() {
+        if(this._observableQuery) {
+            this._observableQuery.stopPolling();
+        }
+    }
+
+    private async _startPollingForApplications() {
+        const query = gql`
+            query {
+                allApplications {
+                    applicationId
+                    name
+                }
+            }
+        `;
+
+        this._observableQuery = this._dataSource.watchQuery<any>({ query });
+
+        this._observableQuery.startPolling(5000);
+        this._observableQuery.subscribe((next) => {
+            if(next.networkStatus === NetworkStatus.ready && next.data){
```

Comment:
This was a WTF moment for me. I didn't understand why the ObservableQuery was being fired twice for each poll. I thought I had screwed up my subscriptions, or that the singleton behaviour wasn't working. What is happening though, is that the observable status is updated with a `loading` NetworkStatus to indicate that it's started the query. That is why there's a check to verify that the status is `ready` before signalling outside that the value has changed.

Reply:
That is interesting. I'll make a note of this - we shouldn't need to do this all over the place.

Reply:
there's also an option to enable/disable notifying on network change with the `notifyOnNetworkStatusChange`flag - but I didn't get this to work. I didn't spend more than a few minutes trying, though. https://www.apollographql.com/docs/react/api/core/ApolloClient/#ApolloClient.query